test/e2e/servers/ory/kratos.spec.ts

```diff
@@ -517,3 +528,106 @@ describe("decoding link header", () => {
     expect(getNextPage(withoutNext)).toBe(undefined)
   })
 })
+
+describe("updates user phone", () => {
+  let newPhone: PhoneNumber
+
+  it("updates user phone", async () => {
+    const { phone, code } = getPhoneAndCodeFromRef("H")
+
+    const { apolloClient, disposeClient } = createApolloClient(defaultTestClientConfig())
+
+    apolloClient.mutate({
+      mutation: USER_LOGIN,
+      variables: { input: { phone, code } },
+    })
+
+    const { phone: adminPhone, code: adminCode } = await getAdminPhoneAndCode()
+
+    const loginResult = await apolloClient.mutate({
+      mutation: USER_LOGIN,
+      variables: { input: { phone: adminPhone, code: adminCode } },
+    })
+
+    await disposeClient()
+
+    const { apolloClient: adminApolloClient, disposeClient: disposeAdminClient } =
+      await createApolloClient(
+        adminTestClientConfig(loginResult.data.userLogin.authToken),
+      )
+
+    const accountDetails = await adminApolloClient.query({
+      query: ACCOUNT_DETAILS_BY_USER_PHONE,
+      variables: { phone },
+    })
+
+    const uid = accountDetails.data.accountDetailsByUserPhone.id
+
+    newPhone = randomPhone()
+    await adminApolloClient.mutate({
+      mutation: USER_UPDATE_PHONE,
+      variables: { input: { phone: newPhone, uid } },
+    })
+
+    const result = await adminApolloClient.query({
+      query: ACCOUNT_DETAILS_BY_USER_PHONE,
+      variables: { phone: newPhone },
+    })
+
+    await disposeAdminClient()
+
+    expect(result.data.accountDetailsByUserPhone.id).toBe(uid)
+  })
+
+  it("updates phone even if new phone is associated with a zero balance account", async () => {
```

Comment:
We may need a failure mode test for this too. Can probably add a test for where the user has an existing balance in the account.

Reply:
That is already being done at the end of that test, perhaps I should improve the title of the test